include/deal.II/fe/mapping_fe_field.h

```diff
@@ -62,50 +62,91 @@ DEAL_II_NAMESPACE_OPEN
  *    dhq.distribute_dofs(fesystem);
  *    Vector<double> eulerq(dhq.n_dofs());
  *    const ComponentMask mask(spacedim, true);
- *    MappingFE<dim,spacedim> map(eulerq, dhq, mask);
+ *    MappingFEField<dim,spacedim> map(eulerq, dhq, mask);
  *    map.update_euler_vector_using_triangulation(eulerq);
```

Comment:
You don't pass it twice. You pass it once only, but if euler vector is zero (as in the default construction of all our vectors), then accessing mapping will always result in a domain collapsed to the single `Point<spacedim>()`, i.e., zero.

Morally, that function belongs to `VectorTools`, not to a member function of MappingFEField. 

You can put whatever you want in euler_vector. If you want to have an interpolation of the Triangulation (that is, mimiking MappingQ1), then the last line conveniently fills euler_vector with the right information.